### PR TITLE
JMB-13 implement location feature

### PR DESCRIPTION
**Pull request by karpushok**
Implement User-Driven Location Setting and Modal-Based Adjustments (JMB-13)

### CODE REVIEW

File: app/components/layout/Header.tsx
[the remark this answers is not in the document]
Comment (hsivritepe) [reply app/components/layout/Header.tsx:1]: That is the reason I have components folder under app folder.

In a Next.js project, whether the components folder is under the root directory or the app directory depends on the specific version of Next.js and your project’s structure. Here’s a breakdown of the options:

1. Under the app Directory (Next.js 13+ with App Router)

If you are using the App Router introduced in Next.js 13, it is common practice to place the components folder under the app directory. This aligns components with the new file-based routing system and allows for easier management of server and client components.

This approach helps you scope components to the app directory and ensures clarity between the app and pages directories if you are using a hybrid setup.

2. Under the root Directory

If you’re using the Pages Router (pre-Next.js 13 or in hybrid projects), or you prefer a more global structure, you can place the components folder in the root directory. This is ideal when your components are shared across multiple parts of the app and not specific to a single directory.

This structure is clean for larger projects where components are reused across different routing systems.

Recommendation
	•	If your project is App Router-only, place the components folder under the app directory for a modular structure.
	•	If you’re using a hybrid setup or the Pages Router, keeping the components folder in the root directory works well for better reusability and separation.

File: app/layout.tsx
Comment (hsivritepe) [inline app/layout.tsx:1]: You need to adjust the components folder text to match the app folder structure.

**Review by hsivritepe — CHANGES_REQUESTED** [review]
Hey,

It is generally good as for the first PR but 2 important things.

- Please put back the components folder under app folder rather than root folder (detailed explanation in the review)

- You need to create the PR to the development branch, not to the live branch

File: app/components/layout/Header.tsx
[the remark this answers is not in the document]
Comment (karpushok) [reply app/components/layout/Header.tsx:1]: Hi, Hakan

Thank you for reviewing my PR. Here are my responses to the comments:

1. Placement of `components` Folder:
   - The `components` folder is already located in the root directory in the `development` branch, as per the existing project structure. I have not moved or relocated it. If you decide to restructure this folder under `app`, I am happy to assist with the migration.

2. Removal of `Header.tsx`:
   - I did not delete or move `Header.tsx`. It remains in `components/layout` as in the `development` branch. My updates to `Header.tsx` only include the addition of location functionality, while retaining its original structure and features.

3. Branch for PR:
   - I acknowledge the mistake in creating the PR against the `live` branch. I will close this PR and reopen it against the `development` branch.

4. Location Functionality:
   - The updates in `Header.tsx` introduce user-driven location functionality. These changes do not alter the overall structure or functionality of the existing components, ensuring compatibility with the current codebase.

Please let me know if you need any further clarifications or additional changes.

Best regards,
Andrei


[development_project_structure.txt](https://github.com/user-attachments/files/18100735/development_project_structure.txt)
[jmb13_project_structure.txt](https://github.com/user-attachments/files/18100736/jmb13_project_structure.txt)

**Comment by karpushok** [comment]
Closing this PR because it was mistakenly opened against the live branch. I will reopen it against the development branch.

**Pull request closed without merging**